src/view.js

```diff
@@ -195,19 +199,51 @@ function startDragOrDraw(d) {
   };
 }
 
-function dragOrDraw(d, withDrag, withDraw) {
+function dragOrDraw(d, withDrag, withDraw, ctrl) {
   return function(e) {
     if ((e.shiftKey || util.isRightButton(e)) && d.drawable.enabled) withDraw(d, e);
-    else if (!d.viewOnly) withDrag(d, e);
+    else if (!d.viewOnly) withDrag(d, e, ctrl);
   };
 }
 
 function bindEvents(ctrl, el, context) {
   var d = ctrl.data;
-  var onstart = startDragOrDraw(d);
-  var onmove = dragOrDraw(d, drag.move, draw.move);
-  var onend = dragOrDraw(d, drag.end, draw.end);
-  var startEvents = ['touchstart', 'mousedown'];
+  var start = startDragOrDraw(d);
+  var move = dragOrDraw(d, drag.move, draw.move);
+  var end = dragOrDraw(d, drag.end, draw.end, ctrl);
+  var mousedown = false;
+
+  document.addEventListener('mousedown', function() {
+    mousedown = true;
+  });
+
+  document.addEventListener('mouseup', function() {
+    mousedown = false;
+  });
+
+  var onstart = function(data, e) {
+    if (pointerSelected(ctrl)) {
+      if (data.type !== 'mousemove') {
+        start(data, e);
+      }
+    } else if (data.type !== 'mousemove' || mousedown) {
+      end(data, e);
+    }
+  };
+
+  var onmove = function(data, e) {
+    if (pointerSelected(ctrl)) {
+      move(data, e);
+    }
+  };
+
+  var onend = function(data, e) {
+    if (pointerSelected(ctrl)) {
+      end(data, e);
+    }
+  };
```

Comment:
Same thing. Put all these behavioral changes behind a boolean setting that's only checked once.